spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/NaiveBayesClassifier.java

```diff
@@ -76,11 +76,25 @@
     @Description("The field on which to set the prediction. It will be of type double.")
     private final String predictionField;
 
-    public Config(String fileSetName, String path, String fieldToClassify, String predictionField) {
+    @Description("The number of features to use in training the model. It must be of type integer and equal to the " +
+                  "number of features used in NaiveBayesTrainer.")
+    private final String numFeatures;
+
+    public Config(String fileSetName, String path, String fieldToClassify, String predictionField, String numFeatures) {
       this.fileSetName = fileSetName;
       this.path = path;
       this.fieldToClassify = fieldToClassify;
       this.predictionField = predictionField;
+      this.numFeatures = numFeatures;
+    }
+
+    public boolean validateNumFeatures() {
+      try {
+        int parsedNumFeatures = Integer.parseInt(numFeatures.trim());
```

Comment:
NumberFormatException is a subclass of IllegalArgumentException, so should be fine to simply allow that to propagate that (don’t catch and rethrow something else)